modules/apps/marketplace/marketplace-js-components-web/src/main/resources/META-INF/resources/js/core/MarketplaceProduct.ts

```diff
@@ -166,6 +166,18 @@ export class MarketplaceProduct {
 		return true;
 	}
 
+	public hasPermissionToInstall(permissions: AppsPermissions) {
+		if (permissions.purchaseAndInstallPaidApps) {
+			return true;
+		}
+
+		if (this.getPriceModel() === 'free') {
```

Comment:
Just in case, as this information is coming from the specification the value can be or not capitalized, it is hard to insure

```suggestion
		if (this.getPriceModel().toLowerCase() === 'free') {
```

Reply:
You can also use the same validation

https://github.com/is-solutions-delivery/liferay-portal/blob/b2be4b08e5d7ab9d13bec0a26a822ef93a03b9b5/modules/apps/marketplace/marketplace-js-components-web/src/main/resources/META-INF/resources/js/core/MarketplaceProduct.ts#L101

Or create a method wrapping this validation and replace and both places